x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRulesIndexService.java

```diff
@@ -439,6 +439,7 @@ private static QueryRulesetResult mapSearchResponseToQueryRulesetList(SearchResp
     }
 
     private static QueryRulesetListItem hitToQueryRulesetListItem(SearchHit searchHit) {
+        // No need to cache the source map as it is not used outside of this method
         final Map<String, Object> sourceMap = searchHit.getSourceAsMap();
```

Comment:
I wonder if all these scenarios deserve a clearer API for parsing some specific fields as opposed to returning the entire map. That way we would enforce that the map is not used outside of these methods and be able to remove all of these comments?

Reply:
We can remove the comments, I just added for the reviewers can check every place where the method is used. I agree the API for SearchHits is not ideal.

Reply:
> we can add a specific methods for these usages already, no? 
Not sure if I follow, can you give an example of the methods your are thinking about?

Reply:
A method that takes the fields needed, and returns their values instead of the entire map

Reply:
The advantage of having that method is that we can isolate usages that require the map, and perhaps caching, as opposed to fire and forget cases that don't require caching.

Reply:
I disagree here because we should not cache in the SearchHit. 

My reasoning is that you normally process a number of hits in a loop. SearchHit caching will persist even when you finish processing that hit, therefore increasing the heap every iteration. 

If a process need caching it should be implemented at that level so it caches on map at a time and can GC the map after the hit has been processed.

Reply:
We should encourage queries to ask just for the data they need so there are less data moving around the system. Having field filtering at this level defeats it.

Reply:
I don't think I follow, and I think we are misunderstanding each other. I'll take a step back. I was looking for a way to avoid all those comments for situations that are safe without caching because they don't need any. If these could be moved to a different API, that would simplify identifying the cases that are affected by the removal of caching etc.

Reply:
My proposal would be to remove those comments.

I only add them because I want reviewers to check that all usages are not getting anything from caching, more the opposite. Once we agree on the change we can remove the comments and then we have the final PR.